pkg/apis/monitoring/v1/types.go

```diff
@@ -970,6 +970,12 @@ func (c *SafeTLSConfig) Validate() error {
 		}
 	}
 
+	if c.Cert != (SecretOrConfigMap{}) && c.KeySecret == nil {
```

Comment:
while we are here: does it make sense to add a unit test for `*SafeTLSConfig.Validate` here in this package? I think we are missing one.